frontend/src/services/AuthService.ts

```diff
@@ -34,7 +34,15 @@ async function login() {
         https://docs.amplify.aws/lib/auth/social/q/platform/js/
         https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
     */
-    Auth.federatedSignIn();
+    // Auth.federatedSignIn();
+
+    const env = JSON.parse(window.localStorage.getItem('env_data') || '');
+
+    window.location.href =
+        `https://${env?.fam_cognito_domain.value}.auth.${env?.fam_cognito_region.value}.amazoncognito.com/` +
+        `oauth2/authorize?identity_provider=${env?.target_env.value.toUpperCase()}-IDIR&` +
```

Comment:
The target_env variable in deployment is actually an injected variable, in the `infrastructure/server/outputs.tf` file, it says as the following, so I assume its value will be "dev" or "test" or "prod"? If so, I just make it to upper case and the identity provider will be "DEV-IDIR" or "TEST-IDIR" or "PROD-IDIR"
```
output "target_env" {
  description = "dev, test, or prod in AWS"
  value = var.target_env
}
```

Reply:
Hmm, this is only temporary at best until BCeID users are added to FAM. And this solution seems like a bit of a hack.
For example, assuming the identity provider name could be avoiding by passing in the IDIR identity provider name into the env_data.